pkg/services/ngalert/metrics/historian.go

```diff
@@ -0,0 +1,45 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type Historian struct {
+	Registerer            prometheus.Registerer
+	TransitionsTotal      *prometheus.CounterVec
+	WriteFailuresTotal    prometheus.Counter
+	ActiveWriteGoroutines prometheus.Gauge
+	PersistDuration       prometheus.Histogram
+}
+
+func NewHistorianMetrics(r prometheus.Registerer) *Historian {
```

Comment:
Shouldn't all these metrics be per org? Maybe the histogram is the only thing I might consider not adding org to it, but IMO, all the others should.

I'd also consider introducing a gauge or a label somewhere that tells us which is the current backend of choice for this instance. Typically this is done with a single metric that gives you information about the system it is connected to, a very nice feature would be to include a gauge with whatever information we can get out of "ping" e.g. `alerting_grafana_historian_info{backend="loki", version="1.3.4"} 1`

Reply:
We don't get really any info about Loki out of ping right now, as we aren't hitting `buildinfo`. This is because `buildinfo` isn't always available... whether it _should_ be is something we can discuss going forward.

I'm only including `backend` for now as that's really all we are able to know currently. If we move to `/buildinfo` in the future we can add more labels at that point.

Otherwise - done to both comments.